### PR TITLE
Change doc requirements to allow Sphinx version 5.0

### PR DESCRIPTION
**Pull request by rikvl**
I ran into the following error when trying to merge PR #218:
```
Sphinx version error:
The sphinx_rtd_theme extension used by this project needs at least Sphinx v5.0; it therefore cannot be built with this version.
Error: Process completed with exit code 2.
```

The present PR allows Sphinx version 5.0

@tristpinsm do you remember the reason for excluding this version (as per your PR #100, over a year ago)?

I tested this by installing Sphinx 5.0.0 and sphinx-rtd-theme 2.0.0 and running the following from inside the main ch_pipeline directory: `sphinx-build -W -b html doc/ doc/_build/html`
That seemed to work without issues

### CODE REVIEW

**Comment by rikvl** [comment]
I changed it to match the caput Sphinx requirement.

Testing the docs build with Sphinx 6.2.1 again seemed to not give any issues

**Review by ljgray — APPROVED** [review]
I'm approving because I don't see anything wrong with this, but I think it's worth waiting for someone who was involved in the decision to pin the version to chime in

**Review by jrs65 — APPROVED** [review]

**Comment by rikvl** [comment]
In a meeting yesterday, Richard suggested we just remove all pins and version constraints and see if that builds. Hopefully that avoids these version-restriction problems from recurring.

It seems to work.

@ljgray, Github tells me I "dismissed your stale review" -- sounds a bit rude, sorry :) Could you (or @tristpinsm) okay the new version?

**Review by ljgray — APPROVED** [review]

**Pull request merged**
